### PR TITLE
Improved window state handling

### PR DESCRIPTION
**Pull request by Razzeee**
As stated in #249 the window handling was off under some circumstances.

This tries to improve it and is working fine with win10 x64.

There might be cases I missed, where this also needs to be done. Just let me know.

### CODE REVIEW

**Comment by jasonblais** [comment]
Perfect, thanks @Razzeee!

Could we have the community's help testing this pull request? Please report back after testing,

Below are the builds you can download for testing:
 - [Windows x64](https://502-43964563-gh.circle-artifacts.com/0//tmp/circle-artifacts.6rHh5lY/mattermost-desktop-win64.zip)
 - [Mac / OS X](https://502-43964563-gh.circle-artifacts.com/0//tmp/circle-artifacts.6rHh5lY/mattermost-desktop-osx.tar.gz)
 - [Linux - x32](https://502-43964563-gh.circle-artifacts.com/0//tmp/circle-artifacts.6rHh5lY/mattermost-desktop-linux-ia32.tar.gz)
 - [Linux - x64](https://502-43964563-gh.circle-artifacts.com/0//tmp/circle-artifacts.6rHh5lY/mattermost-desktop-linux-x64.tar.gz)

Other artifacts [available for download here](https://circleci.com/gh/mattermost/desktop/502#artifacts)

File: src/browser/webview/mattermost.js [hunk 1]
```diff
@@ -118,11 +118,13 @@ notification.override({
...
       }
       else {
-        currentWindow.show();
+        if (mainWindow.isMinimized()) mainWindow.restore()
+        else mainWindow.show();
       }
     }
     else {
-      currentWindow.show();
+      if (mainWindow.isMinimized()) mainWindow.restore()
```
Comment (yuya-oc) [inline src/browser/webview/mattermost.js]: Should use `currentWindow`.

File: src/browser/webview/mattermost.js [hunk 2]
```diff
@@ -118,11 +118,13 @@ notification.override({
         currentWindow.focus();
       }
       else {
-        currentWindow.show();
+        if (mainWindow.isMinimized()) mainWindow.restore()
```
Comment (yuya-oc) [inline src/browser/webview/mattermost.js]: Should use `currentWindow`. Probably should use `else if` because of nested `if` statement.

**Comment by yuya-oc** [comment]
Tested on OS X 10.11.6. The tray icon part works fine. But for clicking notifications, the window is not restored. Please refer to my comments.

**Comment by Razzeee** [comment]
Updated and addressed your comments. Stupid c&p error on my end.

**Comment by yuya-oc** [comment]
Tested on OS X 10.11.6. All works fine.

**Comment by jasonblais** [comment]
@Razzeee 

The repro steps provided in #249 still happen for me on Windows 10

```
1. Open the Windows Desktop App.
2. Maximize the window, then minimize it to the taskbar.
3. Click on a notification or on the tray icon.

Observed: The window comes back to its original size, not maximized.
```

Same thing happens when I install the app for the first time.

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,11 +22,12 @@ Release date: TBD
 - Added an option to toggle the red dot icon for unread messages (default is on).
 
 #### Windows
+- Update Mattermost icon for desktop notifications in Windows 10.
 - Added an option to toogle the red dot icon for unread messages (default is on).
 
 
 ### Bug Fixes
-
+- Fixed but where window would restore the wrong state under some conditions.
```
Comment (jasonblais) [inline CHANGELOG.md]: Proposed changelog entry: `Fixed an issue where the maximized state of the app window was lost in some cases.`

(I think it helps explain what we mean by the word "state")

**Comment by jasonblais** [comment]
My apologies,  I was testing the wrong artifact.

Tested on Windows 10:
 - notifications
 - system tray
 - app launch

and all work when the window is/isn't maximized.

All good to go, have a minor note on the changelog entry @Razzeee

**Comment by jasonblais** [comment]
Anyone from the Linux community who would like to test this pull request? [You can download an artifact here](https://circleci.com/gh/mattermost/desktop/586#artifacts).

File: CHANGELOG.md (hunk 3 above)
Comment (Razzeee) [inline CHANGELOG.md]: Will change, but I think it may also address some other restore issues, that are not that obvious.

**Comment by Razzeee** [comment]
Changed the changelog entry.

**Comment by jnugh** [comment]
Ubuntu 16.04 Unity: Can't reproduce this on master, still works with this PR. However this might be highly related to the used desktop environment and Unity (which I use) might work different compared to Gnome / KDE / ...

**Comment by jasonblais** [comment]
Thanks @jnugh for testing, much appreciated!

Sound like this may not have affected the Linux platform, at least with Unity.

I think we're ready to merge this @yuya-oc

**Comment by yuya-oc** [comment]
Thanks all!

**Pull request merged**